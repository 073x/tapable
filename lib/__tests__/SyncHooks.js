const HookTester = require("./HookTester");
const SyncHook = require("../SyncHook");
const SyncBailHook = require("../SyncBailHook");
const SyncWaterfallHook = require("../SyncWaterfallHook");

describe("SyncHook", () => {
	it("should have to correct behavior", async () => {
		const tester = new HookTester((args) => new SyncHook(args));

		const result = await tester.run(true);

		expect(result).toMatchSnapshot();
	}, 15000)
})

describe("SyncBailHook", () => {
	it("should have to correct behavior", async () => {
		const tester = new HookTester((args) => new SyncBailHook(args));

		const result = await tester.run(true);

		expect(result).toMatchSnapshot();
	}, 15000)
})

describe("SyncWaterfallHook", () => {
	it("should have to correct behavior", async () => {
		const tester = new HookTester((args) => new SyncWaterfallHook(args));

		const result = await tester.run(true);

		expect(result).toMatchSnapshot();
	}, 15000)
})
