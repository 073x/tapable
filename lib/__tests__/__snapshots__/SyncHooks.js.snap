// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncBailHook should have to correct behavior 1`] = `
Object {
  "callIntercepted": Object {
    "type": "return",
    "value": 6,
  },
  "callInterceptedCall1": Array [
    1,
    2,
    3,
  ],
  "callInterceptedCall2": Array [
    1,
    2,
    3,
  ],
  "callInterceptedTap1": Object {
    "fn": 3,
    "name": "sync1",
    "type": "sync",
  },
  "callInterceptedTap2": Object {
    "fn": 3,
    "name": "sync1",
    "type": "sync",
  },
  "callMultipleSync": Object {
    "type": "return",
    "value": 42,
  },
  "callMultipleSyncCalled1": true,
  "callMultipleSyncError": Object {
    "error": "Error in sync2",
  },
  "callMultipleSyncErrorCalled1": true,
  "callMultipleSyncErrorCalled2": true,
  "callMultipleSyncWithArg": Object {
    "type": "return",
    "value": 84,
  },
  "callMultipleSyncWithArgCalled1": 42,
  "callMultipleSyncWithArgFirstReturn": Object {
    "type": "return",
    "value": 84,
  },
  "callMultipleSyncWithArgFirstReturnCalled1": 42,
  "callMultipleSyncWithArgLastReturn": Object {
    "type": "return",
    "value": 85,
  },
  "callMultipleSyncWithArgLastReturnCalled1": 42,
  "callMultipleSyncWithArgLastReturnCalled2": 42,
  "callMultipleSyncWithArgNoReturn": Object {
    "type": "no result",
  },
  "callMultipleSyncWithArgNoReturnCalled1": 42,
  "callMultipleSyncWithArgNoReturnCalled2": 42,
  "callMultipleSyncWithArgs": Object {
    "type": "return",
    "value": 129,
  },
  "callMultipleSyncWithArgsCalled1": Array [
    42,
    43,
    44,
  ],
  "callNone": Object {
    "type": "no result",
  },
  "callNoneWithArg": Object {
    "type": "no result",
  },
  "callSingleSync": Object {
    "type": "return",
    "value": 42,
  },
  "callSingleSyncCalled": true,
  "callSingleSyncWithArg": Object {
    "type": "return",
    "value": 42,
  },
  "callSingleSyncWithArgCalled": 42,
}
`;

exports[`SyncHook should have to correct behavior 1`] = `
Object {
  "callIntercepted": Object {
    "type": "no result",
  },
  "callInterceptedCall1": Array [
    1,
    2,
    3,
  ],
  "callInterceptedCall2": Array [
    1,
    2,
    3,
  ],
  "callInterceptedTap1": Object {
    "fn": 2,
    "name": "sync2",
    "type": "sync",
  },
  "callInterceptedTap2": Object {
    "fn": 3,
    "name": "sync1",
    "type": "sync",
  },
  "callMultipleSync": Object {
    "type": "no result",
  },
  "callMultipleSyncCalled1": true,
  "callMultipleSyncCalled2": true,
  "callMultipleSyncError": Object {
    "error": "Error in sync2",
  },
  "callMultipleSyncErrorCalled1": true,
  "callMultipleSyncErrorCalled2": true,
  "callMultipleSyncWithArg": Object {
    "type": "no result",
  },
  "callMultipleSyncWithArgCalled1": 42,
  "callMultipleSyncWithArgCalled2": 42,
  "callMultipleSyncWithArgFirstReturn": Object {
    "type": "no result",
  },
  "callMultipleSyncWithArgFirstReturnCalled1": 42,
  "callMultipleSyncWithArgFirstReturnCalled2": 42,
  "callMultipleSyncWithArgLastReturn": Object {
    "type": "no result",
  },
  "callMultipleSyncWithArgLastReturnCalled1": 42,
  "callMultipleSyncWithArgLastReturnCalled2": 42,
  "callMultipleSyncWithArgNoReturn": Object {
    "type": "no result",
  },
  "callMultipleSyncWithArgNoReturnCalled1": 42,
  "callMultipleSyncWithArgNoReturnCalled2": 42,
  "callMultipleSyncWithArgs": Object {
    "type": "no result",
  },
  "callMultipleSyncWithArgsCalled1": Array [
    42,
    43,
    44,
  ],
  "callMultipleSyncWithArgsCalled2": Array [
    42,
    43,
    44,
  ],
  "callNone": Object {
    "type": "no result",
  },
  "callNoneWithArg": Object {
    "type": "no result",
  },
  "callSingleSync": Object {
    "type": "no result",
  },
  "callSingleSyncCalled": true,
  "callSingleSyncWithArg": Object {
    "type": "no result",
  },
  "callSingleSyncWithArgCalled": 42,
}
`;

exports[`SyncLoopHook should have to correct behavior 1`] = `
Object {
  "callAsyncInterceptedSync": Object {
    "type": "async",
    "value": undefined,
  },
  "callAsyncInterceptedSyncCalled1": 83,
  "callAsyncInterceptedSyncCalled2": 42,
  "callAsyncInterceptedSyncCalledCall": 1,
  "callAsyncInterceptedSyncCalledLoop": 83,
  "callAsyncInterceptedSyncCalledTap": 125,
  "callAsyncMultipleSync": Object {
    "type": "async",
    "value": undefined,
  },
  "callAsyncMultipleSyncCalled1": 83,
  "callAsyncMultipleSyncCalled2": 42,
  "callAsyncNone": Object {
    "type": "async",
    "value": undefined,
  },
  "callAsyncNoneWithArg": Object {
    "type": "async",
    "value": undefined,
  },
  "callAsyncSingleSync": Object {
    "type": "async",
    "value": undefined,
  },
  "callAsyncSingleSyncCalled": 42,
  "callInterceptedSync": Object {
    "type": "no result",
  },
  "callInterceptedSyncCalled1": 83,
  "callInterceptedSyncCalled2": 42,
  "callInterceptedSyncCalledCall": 1,
  "callInterceptedSyncCalledLoop": 83,
  "callInterceptedSyncCalledTap": 125,
  "callMultipleSync": Object {
    "type": "no result",
  },
  "callMultipleSyncCalled1": 83,
  "callMultipleSyncCalled2": 42,
  "callNone": Object {
    "type": "no result",
  },
  "callNoneWithArg": Object {
    "type": "no result",
  },
  "callSingleSync": Object {
    "type": "no result",
  },
  "callSingleSyncCalled": 42,
  "promiseInterceptedSync": Object {
    "type": "promise",
    "value": undefined,
  },
  "promiseInterceptedSyncCalled1": 83,
  "promiseInterceptedSyncCalled2": 42,
  "promiseInterceptedSyncCalledCall": 1,
  "promiseInterceptedSyncCalledLoop": 83,
  "promiseInterceptedSyncCalledTap": 125,
  "promiseMultipleSync": Object {
    "type": "promise",
    "value": undefined,
  },
  "promiseMultipleSyncCalled1": 83,
  "promiseMultipleSyncCalled2": 42,
  "promiseNone": Object {
    "type": "promise",
    "value": undefined,
  },
  "promiseNoneWithArg": Object {
    "type": "promise",
    "value": undefined,
  },
  "promiseSingleSync": Object {
    "type": "promise",
    "value": undefined,
  },
  "promiseSingleSyncCalled": 42,
}
`;

exports[`SyncWaterfallHook should have to correct behavior 1`] = `
Object {
  "callIntercepted": Object {
    "type": "return",
    "value": 9,
  },
  "callInterceptedCall1": Array [
    1,
    2,
    3,
  ],
  "callInterceptedCall2": Array [
    1,
    2,
    3,
  ],
  "callInterceptedTap1": Object {
    "fn": 2,
    "name": "sync2",
    "type": "sync",
  },
  "callInterceptedTap2": Object {
    "fn": 3,
    "name": "sync1",
    "type": "sync",
  },
  "callMultipleSync": Object {
    "error": "Waterfall hooks must have at least one argument",
  },
  "callMultipleSyncError": Object {
    "error": "Waterfall hooks must have at least one argument",
  },
  "callMultipleSyncWithArg": Object {
    "type": "return",
    "value": 127,
  },
  "callMultipleSyncWithArgCalled1": 42,
  "callMultipleSyncWithArgCalled2": 84,
  "callMultipleSyncWithArgFirstReturn": Object {
    "type": "return",
    "value": 84,
  },
  "callMultipleSyncWithArgFirstReturnCalled1": 42,
  "callMultipleSyncWithArgFirstReturnCalled2": 84,
  "callMultipleSyncWithArgLastReturn": Object {
    "type": "return",
    "value": 85,
  },
  "callMultipleSyncWithArgLastReturnCalled1": 42,
  "callMultipleSyncWithArgLastReturnCalled2": 42,
  "callMultipleSyncWithArgNoReturn": Object {
    "type": "return",
    "value": 42,
  },
  "callMultipleSyncWithArgNoReturnCalled1": 42,
  "callMultipleSyncWithArgNoReturnCalled2": 42,
  "callMultipleSyncWithArgs": Object {
    "type": "return",
    "value": 217,
  },
  "callMultipleSyncWithArgsCalled1": Array [
    42,
    43,
    44,
  ],
  "callMultipleSyncWithArgsCalled2": Array [
    129,
    43,
    44,
  ],
  "callNone": Object {
    "error": "Waterfall hooks must have at least one argument",
  },
  "callNoneWithArg": Object {
    "type": "return",
    "value": 42,
  },
  "callSingleSync": Object {
    "error": "Waterfall hooks must have at least one argument",
  },
  "callSingleSyncWithArg": Object {
    "type": "return",
    "value": 42,
  },
  "callSingleSyncWithArgCalled": 42,
}
`;
